### PR TITLE
Use `metadata.labels["node"]` when restoring `Machine` resources

### LINKED ISSUE #5411: Use the `machine.metadata.label[node]` field instead of `machine.status.node` during worker restoration
**How to categorize this issue?**

/area control-plane-migration
/kind enhancement

**What would you like to be added**:
Currently when restoring the `Worker` resource, machine objects are fetched from the `worker.status.state` field and deployed in the shoot's namespace inside the seed cluster. Afterwards the `machine.status.node` field of these objects is patched so that it corresponds to the node the machine is responsible for. With this the creation of a new node is prevented. once the machine-controller-manager starts:
https://github.com/gardener/gardener/blob/ab3caf94c3fdeb3d6701dc4c8addb485e56a1cae/extensions/pkg/controller/worker/genericactuator/actuator_restore.go#L130-L146

However, there is also a `machine.metadata.labels[node]` field present in the machine object, which also points to the node that the machine object is responsible for. The restoration logic would be simpler if we relied on the field from the labels instead of relying on `machine.status.node`.

**Additional info**
I tested a few control plane migrations by only relying on `machine.metadata.labels[node]` and they finished successfully (without recreating the nodes) but I would like a confirmation from @gardener/mcm-maintainers if this would work in all cases.

### PR DESCRIPTION
**Pull request by plkokanov**
**How to categorize this PR?**

/area control-plane-migration
/kind enhancement

**What this PR does / why we need it**:
This PR removes the additional `PATCH` that was setting the `status.node` of `Machine` resources when restoring them. Currently, the `metadata.labels[node]` field in Machine objects provides the same information as `status.node` and as per this comment: https://github.com/gardener/gardener/issues/5411#issuecomment-1031039104 `status.node` can even be removed from the `Machine` resources as it is not really necessary.

**Which issue(s) this PR fixes**:
Fixes #5411 

**Special notes for your reviewer**:
/cc @himanshu-kun @AxiomSamarth 

**Release note**:

```other developer
Removed unnecessary `PATCH` to `machine.status.node` during restoration of machine objects.
```

### CODE REVIEW

File: extensions/pkg/controller/worker/state_actuator.go
Comment (shafeeqes) [inline extensions/pkg/controller/worker/state_actuator.go:195]: Is it necessary to explicitly set this here?

**Comment by shafeeqes** [comment]
Had a combined look with @ary1992, looks good.
Just few doubts @plkokanov, where is this node label used, while restoring?
We are getting rid of only patching the label while restoring, but we are still making use of this status field?
Like here, https://github.com/gardener/machine-controller-manager/blob/5ddf125802ec4a7e1596c63260441cc173b13d23/pkg/util/provider/machinecontroller/machine.go#L517. I don't know the details but will need to be adapted?
As for this change,
/lgtm

File: extensions/pkg/controller/worker/state_actuator.go
Comment (shafeeqes) [reply extensions/pkg/controller/worker/state_actuator.go:195]: Okay, this will be reconciled by MCM after migration?

Comment (plkokanov) [reply extensions/pkg/controller/worker/state_actuator.go:195]: We set this explicitly here, because adding the `Status` to the `Worker` state is not necessary. Without this `Machine.Status` would most likely not be empty.

**Comment by plkokanov** [comment]
> Had a combined look with @ary1992, looks good. Just few doubts @plkokanov, where is this node label used, while restoring? We are getting rid of only patching the label while restoring, but we are still making use of this status field? Like here, https://github.com/gardener/machine-controller-manager/blob/5ddf125802ec4a7e1596c63260441cc173b13d23/pkg/util/provider/machinecontroller/machine.go#L517. I don't know the details but will need to be adapted?

The `node` label is necessary to make sure that MCM has already created a VM for a `Machine` resource and if the MCM driver (the implementation that will actually create the machine) does not implement `GetMachineStatus()`. You can check this here: https://github.com/gardener/machine-controller-manager/blob/4215b07f21f7b95a9336597073d24f88cfc9a3b5/pkg/util/provider/machinecontroller/machine.go#L374

I could not understand very well what you mean with the code that you linked. The problem there is that the status will shortly (around 15s, until the next reconciliation) contain the following text: "Creating machine on cloud provider", when the machine actually already exists. But I assumed that this could be adapted with https://github.com/gardener/machine-controller-manager/issues/389
@himanshu-kun please correct me if I am wrong.

**Comment by shafeeqes** [comment]
> I could not understand very well what you mean with the code that you linked

Okay, I misunderstood that code.
Thanks for explaining.

Would you like to have a look, before we merge this @himanshu-kun ?

**Comment by himanshu-kun** [comment]
> > Had a combined look with @ary1992, looks good. Just few doubts @plkokanov, where is this node label used, while restoring? We are getting rid of only patching the label while restoring, but we are still making use of this status field? Like here, https://github.com/gardener/machine-controller-manager/blob/5ddf125802ec4a7e1596c63260441cc173b13d23/pkg/util/provider/machinecontroller/machine.go#L517. I don't know the details but will need to be adapted?
> 
> The `node` label is necessary to make sure that MCM has already created a VM for a `Machine` resource and if the MCM driver (the implementation that will actually create the machine) does not implement `GetMachineStatus()`. You can check this here: https://github.com/gardener/machine-controller-manager/blob/4215b07f21f7b95a9336597073d24f88cfc9a3b5/pkg/util/provider/machinecontroller/machine.go#L374
> 
> I could not understand very well what you mean with the code that you linked. The problem there is that the status will shortly (around 15s, until the next reconciliation) contain the following text: "Creating machine on cloud provider", when the machine actually already exists. But I assumed that this could be adapted with [gardener/machine-controller-manager#389](https://github.com/gardener/machine-controller-manager/issues/389) @himanshu-kun please correct me if I am wrong.

Yes @plkokanov , we will not depend on `status.node` to have the `Creating machine on cloud provider` message in status once PR for this issue is raised.

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Comment by timebertt** [comment]
/hold
PR is blocking tide

**Pull request merged**
